src/lib.rs

```diff
@@ -446,7 +445,7 @@ impl<S: Storage + Send + Sync + 'static> SelfEncryptor<S> {
             }
         }
         self.prepare_window(data.len() as u64, position);
-        for i in 0..data.len() {
+        for (i, _) in data.iter().enumerate() {
             self.sequencer[position as usize + i] = data[i];
```

Comment:
How about:

    for (i, &byte) in data.iter().enumerate() {
        self.sequencer[position as usize + i] = byte;